dipy/reconst/sfm.py

```diff
@@ -420,7 +420,24 @@ def design_matrix(self):
         return sfm_design_matrix(self.gtab, self.sphere, self.response,
                                  'signal')
 
-    def fit(self, data, mask=None):
+    def fit_solver2voxels(self, isopredict, vox_data, vox, parallel=False):
+        # In voxels in which S0 is 0, we just want to keep the
+        # parameters at all-zeros, and avoid nasty sklearn errors:
+        if not (np.any(~np.isfinite(vox_data)) or np.all(
+                vox_data == 0)):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```

Comment:
what is the goal of this?

Reply:
Was in the original implementation I think? My guess would be that it suppresses icky sklearn errors on the solver fit.